lib/router.dart

```diff
@@ -1,75 +1,123 @@
+import 'dart:async';
+
 import 'package:flutter/material.dart';
 import 'package:go_router/go_router.dart';
 
 import 'package:solaris_structure_1/main.dart';
 import 'package:solaris_structure_1/screens/hub/hub_screen.dart';
 import 'package:solaris_structure_1/screens/home/home_screen.dart';
+import 'package:solaris_structure_1/screens/login/login_screen.dart';
 import 'package:solaris_structure_1/screens/transfer/transfer_screen.dart';
 
-final _rootNavigatorKey = GlobalKey<NavigatorState>();
-final _shellNavigatorKey = GlobalKey<NavigatorState>();
+import 'cubits/login_cubit/login_cubit.dart';
 
-final router = GoRouter(
-  navigatorKey: _rootNavigatorKey,
-  initialLocation: '/home',
-  debugLogDiagnostics: true,
-  routes: [
-    ShellRoute(
-      navigatorKey: _shellNavigatorKey,
-      builder: (context, state, child) {
-        return AppScaffold(child: child);
-      },
+class AppRouter {
+  final _rootNavigatorKey = GlobalKey<NavigatorState>();
+  final _shellNavigatorKey = GlobalKey<NavigatorState>();
+
+  final LoginCubit loginCubit;
+  AppRouter(this.loginCubit);
+
+  late final router = GoRouter(
+      navigatorKey: _rootNavigatorKey,
+      initialLocation: '/home',
```

Comment:
I propose to replace these [magic strings](https://deviq.com/antipatterns/magic-strings) with constants.

Reply:
@ilie-lupu-ts @szaszgyozo

Reply:
Fixed in [this](https://github.com/ivoryio/SolarisDemoApp/pull/3) PR.